### PR TITLE
Improve type annotations in spiderfoot/target.py

### PR DESCRIPTION
**Pull request by RA80533**

### CODE REVIEW

File: spiderfoot/target.py [hunk 1]
```diff
@@ -1,22 +1,27 @@
+import typing
+
 import netaddr
 
 
+TargetAlias = typing.Dict[typing.Literal["type", "value"], str]
```
Comment (RA80533) [inline spiderfoot/target.py:6]:
```suggestion
TargetAlias = typing.TypedDict("TargetAlias", {"type": str, "value": str})
```

**Comment by smicallef** [comment]
Thanks for the contribution!

File: spiderfoot/target.py [hunk 2]
```diff
@@ -14,9 +14,9 @@ class SpiderFootTarget():
     _validTypes = ["IP_ADDRESS", 'IPV6_ADDRESS', "NETBLOCK_OWNER", "NETBLOCKV6_OWNER", "INTERNET_NAME",
                    "EMAILADDR", "HUMAN_NAME", "BGP_AS_OWNER", 'PHONE_NUMBER', "USERNAME",
```
Comment (rrajrajakrk-creator) [inline spiderfoot/target.py]: 9771042810

File: spiderfoot/target.py [hunk 3]
```diff
@@ -1,6 +1,11 @@
+import typing
```
Comment (rrajrajakrk-creator) [inline spiderfoot/target.py]: 97701402810

**Pull request merged**